src/explorer-ui.js

```diff
@@ -2,63 +2,36 @@ import React, { useState, useEffect, useContext } from 'react';
 
 import { Link, LinkElement } from './link';
 import Resource from './resource';
-import { request } from './lib/request';
 import { LocationContext } from './location';
 
-const ExplorerUI = ({ homeUrl }) => {
-    const location = useContext(LocationContext);
-    const [result, setResult] = useState([]);
-    const [links, setLinks] = useState([]);
-    const [resourceLinks, setResourceLinks] = useState([]);
-
-    const loadTopLevel = () => {
-        const fetchDocument = async (url) => {
-            const response = await request(url);
-            setResourceLinks(Link.parseLinks(response.links));
-        };
-
-        fetchDocument(homeUrl);
-    };
-
-    const updateDocument = () => {
-        const fetchDocument = async (url) => {
-            const response = await request(url);
-            setLinks(Link.parseLinks(response.links));
-            setResult(response);
-        };
-
-        fetchDocument(location.locationUrl);
-    };
-
+const ExplorerUI = () => {
+    const { locationUrl, document, onEntryPoint } = useContext(LocationContext);
+    const [ entryPointLinks, setEntryPointLinks ] = useState({});
+    const parsedLinks = document.links ? Link.parseLinks(document.links) : {};
 
     useEffect(() => {
-        if (location.locationUrl === homeUrl) {
-            loadTopLevel();
-        }
-        else {
-            updateDocument();
-        }
-    }, [location.locationUrl]);
+      if (onEntryPoint) setEntryPointLinks(parsedLinks);
+    }, [onEntryPoint]);
```

Comment:
`onEntryPoint` will change on each location change, wouldn't it be better to just make this something that won't ever change, like `homeUrl` ? You're already checking its boolean inside the effect.